### PR TITLE
Creating get activity route✅

### PR DESCRIPTION
**Pull request by LucaPinheiro**

### CODE REVIEW

File: src/modules/get_activity/app/get_activity_presenter.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { GetActivityUsecase } from "./get_activity_usecase";
+import { Repository } from "../../../core/repositories/Repository";
+
+const repository = new Repository({ activity_repo: true });
+
+const usecase = new GetActivityUsecase(repository.UserRepo, repository.ActivityRepo);
+const controller = new GetActivityController(usecase);
+
+export const handler = async (event: any, context: any) => {
+
```
Comment (FelipeCarillo) [inline src/modules/get_activity/app/get_activity_presenter.ts]: Hmmm?

File: src/modules/get_activity/app/get_activity_controller.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  public async execute(request: HttpRequest): Promise<HttpResponse> {
+    try {
+      if (!request) {
+        throw new InvalidRequest();
+      }
+      if (!request.headers) {
+        throw new InvalidRequest("Headers");
+      }
+
```
Comment (FelipeCarillo) [inline src/modules/get_activity/app/get_activity_controller.ts:33]: Check if the body isn't null

File: src/modules/get_activity/app/get_activity_usecase.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  public user_repo: IUserRepo;
+  public activity_repo: IActivityRepo;
+
+  constructor(user_repo: IUserRepo, activity_repo: IActivityRepo) {
+    this.token_auth = new TokenAuth();
+    this.user_repo = user_repo;
+    this.activity_repo = activity_repo;
+  }
+
+  async execute(headers: { [key: string]: any }, body: { [key: string]: any }) {
```
Comment (FelipeCarillo) [inline src/modules/get_activity/app/get_activity_usecase.ts]: Change the var name body to queryStringParameters

**Review by FelipeCarillo — CHANGES_REQUESTED** [review]

**Review by LucaPinheiro — COMMENTED** [review]
OK

File: src/modules/get_activity/app/get_activity_presenter.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { HttpRequest } from "../../../core/helpers/http/http_codes";
+import { GetActivityController } from "./get_activity_controller";
+import { GetActivityUsecase } from "./get_activity_usecase";
+import { Repository } from "../../../core/repositories/Repository";
+
+const repository = new Repository({ activity_repo: true });
```
Comment (FelipeCarillo) [inline src/modules/get_activity/app/get_activity_presenter.ts]: Coloca user_repo aqui!

File: src/modules/get_activity/app/get_activity_usecase.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      });
+
+    const user = await this.user_repo.get_user(user_id);
+    if (!user) {
+      throw new UserNotAuthenticated();
+    }
+
+    const activity = await this.activity_repo.get_activity(queryStringParameters.activity_id);
+    if (!activity) {
+      throw new NotFound("Activity not found");
```
Comment (FelipeCarillo) [inline src/modules/get_activity/app/get_activity_usecase.ts]: NotFoundError, ai no controle você chama o NotFound

File: src/modules/get_activity/app/get_activity_usecase.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    if (user.user_type === UserTypeEnum.STUDENT) {
+      const check_user_enrolled = this.activity_repo.check_activity_enrolled_by_user(user_id, activity.id);
```
Comment (FelipeCarillo) [inline src/modules/get_activity/app/get_activity_usecase.ts]: faltou o await

File: src/modules/get_activity/app/get_activity_usecase.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+      const condition = ActivityStatusEnum.CANCELED === activity.status_activity || (!check_user_enrolled && [ActivityStatusEnum.ON_HOLD, ActivityStatusEnum.ENDED].includes(activity.status_activity))
+      if (condition) {
+        throw new NotFound("Activity not found");
+      }
+    }
```
Comment (FelipeCarillo) [inline src/modules/get_activity/app/get_activity_usecase.ts:88]: Coloca uma lógica que se o usuário for admin ou moderador, para retornar os usuários que aplicaram!

**Review by FelipeCarillo — CHANGES_REQUESTED** [review]

File: src/modules/get_activity/app/get_activity_controller.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    } catch (error) {
+      if (error instanceof InvalidRequest) {
+        return new BadRequest(error.message);
+      }
+      if (error instanceof UserNotAuthenticated) {
+        return new Unauthorized(error.message);
+      }
+      if (error instanceof MissingParameter) {
+        return new ParameterError(error.message);
+      }
```
Comment (FelipeCarillo) [inline src/modules/get_activity/app/get_activity_controller.ts:50]: Implement the NotFoundError to return the NotFound

File: src/modules/get_activity/app/get_activity_presenter.ts (hunk 4 above)
Comment (LucaPinheiro) [reply src/modules/get_activity/app/get_activity_presenter.ts]: fixed

File: src/modules/get_activity/app/get_activity_usecase.ts [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+    const user = await this.user_repo.get_user(user_id);
+    if (!user) {
+      throw new UserNotAuthenticated();
+    }
+
+    const activity = await this.activity_repo.get_activity(queryStringParameters.activity_id);
+    if (!activity) {
+      throw new NotFound("Activity not found");
+    }
```
Comment (FelipeCarillo) [inline src/modules/get_activity/app/get_activity_usecase.ts:71]: Pega os usuários que aplicaram

**Review by FelipeCarillo — CHANGES_REQUESTED** [review]

File: src/modules/get_activity/app/get_activity_usecase.ts (hunk 5 above)
Comment (LucaPinheiro) [reply src/modules/get_activity/app/get_activity_usecase.ts]: finish

File: src/modules/get_activity/app/get_activity_controller.ts (hunk 8 above)
Comment (LucaPinheiro) [reply src/modules/get_activity/app/get_activity_controller.ts:50]: Finish

File: src/modules/get_activity/app/get_activity_usecase.ts (hunk 9 above)
Comment (LucaPinheiro) [reply src/modules/get_activity/app/get_activity_usecase.ts:71]: finish

File: src/modules/get_activity/app/get_activity_controller.ts (hunk 8 above)
Comment (FelipeCarillo) [reply src/modules/get_activity/app/get_activity_controller.ts:50]: You need to implement the instance error here, so the controller can get the NotFoundError

File: src/modules/get_activity/app/get_activity_usecase.ts [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+          activity.id
+        );
+      const condition =
+        ActivityStatusEnum.CANCELED === activity.status_activity ||
+        (!check_user_enrolled &&
+          [ActivityStatusEnum.ON_HOLD, ActivityStatusEnum.ENDED].includes(
+            activity.status_activity
+          ));
+      if (condition) {
+        throw new NotFound("Activity not found");
```
Comment (FelipeCarillo) [inline src/modules/get_activity/app/get_activity_usecase.ts]: NotFoundError here

File: src/modules/get_activity/app/get_activity_usecase.ts [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+      }
+    }
+
+    const enrolled_users = await this.activity_repo.get_users_assigned_to_activity(
```
Comment (FelipeCarillo) [inline src/modules/get_activity/app/get_activity_usecase.ts]: Here is only with admin or moderator users, students can't have it

**Review by FelipeCarillo — CHANGES_REQUESTED** [review]

File: src/modules/get_activity/app/get_activity_controller.ts (hunk 8 above)
Comment (LucaPinheiro) [reply src/modules/get_activity/app/get_activity_controller.ts:50]: ok

File: src/modules/get_activity/app/get_activity_usecase.ts (hunk 10 above)
Comment (LucaPinheiro) [reply src/modules/get_activity/app/get_activity_usecase.ts]: ok

**Pull request merged**